### PR TITLE
MMCA-4111 : Spike - Compilation warning for Repo - 'customs-manage-au…

### PR DESCRIPTION
**Pull request by sameerhmrc**
JIRA Link : https://jira.tools.tax.service.gov.uk/browse/MMCA-4111

### CODE REVIEW

**Comment by tyagikumarvineet** [comment]
Below 2 comments have been added here as file related to the generated warnings is not in the list of changed files.

1. Could you please add below (**after** scalacOptions += "-P:silencer:globalFilters=possible missing interpolator: detected interpolated identifier `\\$date`",) in build.sbt to fix the below warning?

scalacOptions ++= Seq("-Ypatmat-exhaust-depth", "40"),


[warn] /Users/vineet.tyagi/hmrc/codebase/customs-manage-authorities-frontend/app/models/package.scala:118:7: Exhaustivity analysis reached max recursion depth, not all missing cases are reported.
[warn] (Please try with scalac -Ypatmat-exhaust-depth 40 or -Ypatmat-exhaust-depth off.)
[warn]       (path.path, jsValue) match {
[warn]       ^
[warn] one warning found

2. Once above is done, please add below (after the last case statement) in remove method of models.package. Write a test, if it takes longer then leave it

case _ => JsError(s"cannot remove a key on $jsValue")

File: build.sbt [hunk 1]
```diff
@@ -13,7 +13,6 @@ lazy val root = (project in file("."))
   .disablePlugins(JUnitXmlReportPlugin) //Required to prevent https://github.com/scalatest/scalatest/issues/1427
```
Comment (tyagikumarvineet) [inline build.sbt:13]: 1. Could you please add below (after scalacOptions += "-P:silencer:globalFilters=possible missing interpolator: detected interpolated identifier \\$date",) to fix the below warning?
scalacOptions ++= Seq("-Ypatmat-exhaust-depth", "40"),

[warn] /Users/vineet.tyagi/hmrc/codebase/customs-manage-authorities-frontend/app/models/package.scala:118:7: Exhaustivity analysis reached max recursion depth, not all missing cases are reported.
[warn] (Please try with scalac -Ypatmat-exhaust-depth 40 or -Ypatmat-exhaust-depth off.)
[warn] (path.path, jsValue) match {
[warn] ^
[warn] one warning found

2. Once above is done, please add below (after the last case statement) in remove method of models.package. Write a test, if it takes longer then leave it
case _ => JsError(s"cannot remove a key on $jsValue")

Comment (sameerhmrc) [reply build.sbt:13]: Changes done

File: app/connectors/CustomsFinancialsConnector.scala [hunk 2]
```diff
@@ -41,25 +41,25 @@ class CustomsFinancialsConnector @Inject()(
       AccountsRequestCommon.generate, AccountsRequestDetail(eori, None, None, None)))
 
     httpClient.POST[AccountsAndBalancesRequestContainer,
-      AccountsAndBalancesResponseContainer](baseUrl + context + "/eori/accounts/", request).map(_.toCdsAccounts(eori))
+      AccountsAndBalancesResponseContainer](baseUrl.toString + context.toString + "/eori/accounts/", request).map(_.toCdsAccounts(eori))
```
[the remark this answers is not in the document]
Comment (sameerhmrc) [reply app/connectors/CustomsFinancialsConnector.scala]: Changes done

File: app/models/package.scala [hunk 3]
```diff
@@ -135,6 +135,8 @@ package object models {
                     JsSuccess(Json.arr())
                   case _: RecursiveSearch =>
                     JsError("recursive search is not supported")
+                  case _ =>
```
Comment (tyagikumarvineet) [inline app/models/package.scala:138]: Could you please move this statement from here (to, after line 147) and make the last case statement for (path.path, jsValue) match ?

Comment (sameerhmrc) [reply app/models/package.scala]: Changes done

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**